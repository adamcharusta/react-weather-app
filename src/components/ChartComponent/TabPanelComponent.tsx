import React from 'react';
import { Box, Typography } from '@mui/material';

type TabPanelComponentPropsType = {
  children?: React.ReactNode;
  index: number;
  value: number;
};

const TabPanelComponent = ({ children, value, index, ...other }: TabPanelComponentPropsType) => {
  return (
    <div
      role='tabpanel'
      hidden={value !== index}
      id={`simple-tabpanel-${index}`}
      aria-labelledby={`simple-tab-${index}`}
      {...other}
    >
      {value === index && (
        <Box sx={{ p: 3 }}>
          <Typography>{children}</Typography>
        </Box>
      )}
    </div>
  );
};

export default TabPanelComponent;
