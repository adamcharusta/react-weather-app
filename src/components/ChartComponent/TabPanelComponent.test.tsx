import React from 'react';
import { render } from '@testing-library/react';
import TabPanelComponent from './TabPanelComponent';

describe('TabPanelComponent', () => {
  it('snapshot', () => {
    const component = render(
      <TabPanelComponent index={0} value={0}>
        Hello There!
      </TabPanelComponent>,
    );

    expect(component.asFragment()).toMatchSnapshot();
  });
});
