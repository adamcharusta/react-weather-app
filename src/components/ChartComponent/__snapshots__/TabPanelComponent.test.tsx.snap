// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabPanelComponent snapshot 1`] = `
<DocumentFragment>
  <div
    aria-labelledby="simple-tab-0"
    id="chart-tabpanel-0"
    role="tabpanel"
  >
    <div
      class="MuiBox-root css-0"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
      >
        Hello There!
      </p>
    </div>
  </div>
</DocumentFragment>
`;
